runtime/lua/vim/lsp/util.lua

```diff
@@ -203,6 +203,13 @@ local function remove_unmatch_completion_items(items, prefix)
   end, items)
 end
 
+-- Acording to LSP spec, if the client set "completionItemKind.valueSet",
+-- the client must handle it properly even if it receives a value outside the specification.
+-- https://microsoft.github.io/language-server-protocol/specifications/specification-current/#textDocument_completion
+function M.get_completion_item_kind_name(completion_item_kind)
```

Comment:
Not sure if it's a good idea to clutter the public API with small methods like this. Is this motivated by the tests or with extensibility in mind?

Reply:
Well, it might be useful to allow for customizable kind names (per server)?

(Since the built-in LSP is supposed to be low-level, having a larger, composable, API of small functions would be helpful.)

Reply:
> Since the built-in LSP is supposed to be low-level, having a larger, composable, API of small functions would be helpful

I agree that it would be helpful, but it is a tradeoff. Once a method is public it becomes harder to change it because of BWC. (At least once 0.5 is out) - so I think being a bit conservative about exposing methods may be beneficial.

Just my opinion, not sure how the maintainers see that.